### PR TITLE
Feature/client interfaces and dummies php 8

### PR DESCRIPTION
**Pull request by sleipi**
see #31

### CODE REVIEW

File: src/AbstractClient.php [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+    public function __construct(
+        ?ClientInterface $client = null,
+        string $baseUrl = ''
+    ) {
+        $this->client = $client;
+        $this->baseUrl = $baseUrl;
+    }
+
+    public function getClient(): ?ClientInterface
```
Comment (sleipi) [inline src/AbstractClient.php:22]: Why exposing the ClientInterface in here or do you see any use case to send this class around.

Would suggest to keep it private

File: src/AbstractClient.php [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    /**
+     * @throws \GuzzleHttp\Exception\GuzzleException
+     */
+    protected function sendToServer(
+        string $url = '',
+        string $body = '',
+        string $restType = 'PUT',
+        array $headers = ['Content-Type' => "application/x-www-form-urlencoded"]
+    ): void {
+        $this->client?->request(
```
Comment (sleipi) [inline src/AbstractClient.php:36]: Do use php8 features in here - this client need to support php7.4 

see composer.json
"php": "^7.4 | ^8.0",

File: src/AbstractClient.php [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+use GuzzleHttp\ClientInterface;
+
+abstract class AbstractClient
+{
+    private ?ClientInterface $client;
+    protected string $baseUrl = '';
+
+    public function __construct(
+        ?ClientInterface $client = null,
```
Comment (sleipi) [inline src/AbstractClient.php:15]: Why is the ClientInterface nullable?

The Client is required in this class. w/o a client the whole instance does make sense.
And if there is a good reason why it should nullable so please decide if nullable or optimal.

Even the baseUrl should not be empty

**Comment by sleipi** [comment]
CI is failing and need a some love :)

File: src/AbstractClient.php (hunk 1 above)
Comment (Milanowicz) [reply src/AbstractClient.php]: Yeah, you right and it's gone.

File: src/AbstractClient.php (hunk 2 above)
Comment (Milanowicz) [reply src/AbstractClient.php]: Yeah, it's was the first try ....

File: src/AbstractClient.php (hunk 3 above)
Comment (Milanowicz) [reply src/AbstractClient.php:15]: I did this for @drieschel and not for me. Why he need it ......

**Comment by Milanowicz** [comment]
Did some love for PHP under the Eight :+1:

**Comment by Milanowicz** [comment]
I reduze lines by Infection :+1:

**Pull request merged**